deploy/terraform/terraform-units/modules/sap_system/common_infrastructure/nsgs.tf

```
@@ -0,0 +1,81 @@
+/*-----------------------------------------------------------------------------8
+|                                                                              |
+|                                 NSG                                   |
+|                                                                              |
++--------------------------------------4--------------------------------------*/
+
+# Creates SAP db subnet nsg
+resource "azurerm_network_security_group" "db" {
+  count               = local.enable_db_deployment ? (local.sub_db_nsg_exists ? 0 : 1) : 0
+  name                = local.sub_db_nsg_name
+  resource_group_name = local.rg_exists ? data.azurerm_resource_group.resource_group[0].name : azurerm_resource_group.resource_group[0].name
+  location            = local.rg_exists ? data.azurerm_resource_group.resource_group[0].location : azurerm_resource_group.resource_group[0].location
+}
+
+# Imports the SAP db subnet nsg data
+data "azurerm_network_security_group" "db" {
+  count               = local.enable_db_deployment ? (local.sub_db_nsg_exists ? 1 : 0) : 0
+  name                = split("/", local.sub_db_nsg_arm_id)[8]
+  resource_group_name = split("/", local.sub_db_nsg_arm_id)[4]
+}
+
+# Associates SAP db nsg to SAP db subnet
+resource "azurerm_subnet_network_security_group_association" "db" {
+  count                     = local.enable_db_deployment ? signum((local.sub_db_exists ? 0 : 1) + (local.sub_db_nsg_exists ? 0 : 1)) : 0
+  subnet_id                 = local.sub_db_exists ? data.azurerm_subnet.db[0].id : azurerm_subnet.db[0].id
+  network_security_group_id = local.sub_db_nsg_exists ? data.azurerm_network_security_group.db[0].id : azurerm_network_security_group.db[0].id
+}
+
+# Creates SAP admin subnet nsg
+resource "azurerm_network_security_group" "admin" {
+  count               = ! local.sub_admin_nsg_exists && local.enable_admin_subnet ? 0 : 1
```

Should be 1 : 0 not 0 : 1:
```suggestion
  count               = ! local.sub_admin_nsg_exists && local.enable_admin_subnet ? 1 : 0
```